cea/interfaces/dashboard/api/project.py

```diff
@@ -20,18 +28,30 @@
     'scenarios': fields.List(fields.String, description='Name of Current Scenario')
 })
 
+NEW_PROJECT_MODEL = api.model('New Project', {
+    'name': fields.String(description='Name of Project'),
+    'path': fields.String(description='Path of Project')
+})
+
 
 @api.route('/')
 class Project(Resource):
     @api.marshal_with(PROJECT_MODEL)
     def get(self):
-        config = cea.config.Configuration()
+        config = current_app.cea_config
         name = os.path.basename(config.project)
-        return {'name': name, 'path': config.project, 'scenario': config.scenario_name, 'scenarios': config.get_parameter('general:scenario-name')._choices}
+        if not os.path.exists(config.project):
+            abort(400, 'Project path does not exist')
+        if not os.path.exists(config.scenario):
+            config.scenario_name = ''
+            config.save()
+        return {'name': name, 'path': config.project, 'scenario': config.scenario_name,
+                'scenarios': config.get_parameter('general:scenario-name')._choices}
 
     @api.doc(body=PROJECT_PATH_MODEL, responses={200: 'Success', 400: 'Invalid Path given'})
-    def post(self):
-        config = cea.config.Configuration()
+    def put(self):
```

Comment:
are you sure this isn't the wrong way round? `PUT` should be used to create a new one, `POST` to update an existing project?

Reply:
yeah, `POST` is to **create** and `PUT` is to **update/replace**

Reply:
@reyery you're right! i had this wrong in my head all these years! thank you :)